components/net_connect/ppp_connect.c

```diff
@@ -0,0 +1,276 @@
+        vTaskDelete(NULL);
+        return;
+    }
+    uart_event_t event;
+    esp_event_handler_register(IP_EVENT, IP_EVENT_PPP_GOT_IP, esp_netif_action_connected, s_netif);
+    esp_netif_action_start(s_netif, 0, 0, 0);
+    esp_netif_action_connected(s_netif, 0, 0, 0);
+    while (!s_stop_task) {
+        if (xQueueReceive(event_queue, &event, pdMS_TO_TICKS(1000)) == pdTRUE) {
+            if (event.type == UART_DATA) {
+                size_t len;
+                uart_get_buffered_data_len(UART_NUM_1, &len);
+                if (len) {
+                    len = uart_read_bytes(UART_NUM_1, buffer, BUF_SIZE, 0);
+                    ESP_LOG_BUFFER_HEXDUMP(TAG, buffer, len, ESP_LOG_VERBOSE);
+                    esp_netif_receive(s_netif, buffer, len, NULL);
+                }
+            } else {
+                ESP_LOGW(TAG, "Received UART event: %d", event.type);
+            }
+        }
+    }
+    free(buffer);
+    vTaskDelete(NULL);
+}
+
+#endif // CONNECT_PPP_DEVICE
+
+esp_err_t net_connect_ppp_connect(void)
+{
+    ESP_LOGI(TAG, "Start example_connect.");
+
+#if CONFIG_NET_CONNECT_PPP_DEVICE_USB
+    ESP_LOGI(TAG, "USB initialization");
+    const tinyusb_config_t tusb_cfg = {
+        .device_descriptor = NULL,
+        .string_descriptor = NULL,
+        .external_phy = false,
+        .configuration_descriptor = NULL,
+    };
+
+    ESP_ERROR_CHECK(tinyusb_driver_install(&tusb_cfg));
+
+    tinyusb_config_cdcacm_t acm_cfg = {
+        .usb_dev = TINYUSB_USBDEV_0,
+        .cdc_port = TINYUSB_CDC_ACM_0,
+        .callback_rx = &cdc_rx_callback,
+        .callback_rx_wanted_char = NULL,
+        .callback_line_state_changed = NULL,
+        .callback_line_coding_changed = NULL
+    };
+
+    ESP_ERROR_CHECK(tusb_cdc_acm_init(&acm_cfg));
+    /* the second way to register a callback */
+    ESP_ERROR_CHECK(tinyusb_cdcacm_register_callback(
+                        TINYUSB_CDC_ACM_0,
+                        CDC_EVENT_LINE_STATE_CHANGED,
+                        &line_state_changed));
+#endif // CONFIG_NET_CONNECT_PPP_DEVICE_USB
+
+    s_event_group = xEventGroupCreate();
+
+    ESP_ERROR_CHECK(esp_event_handler_register(IP_EVENT, ESP_EVENT_ANY_ID, on_ip_event, NULL));
+
+    esp_netif_inherent_config_t base_netif_cfg = ESP_NETIF_INHERENT_DEFAULT_PPP();
+    base_netif_cfg.if_desc = NET_CONNECT_NETIF_DESC_PPP;
+    esp_netif_config_t netif_ppp_config = { .base = &base_netif_cfg,
+                                            .driver = ppp_driver_cfg,
+                                            .stack = ESP_NETIF_NETSTACK_DEFAULT_PPP
+                                          };
+
+    s_netif = esp_netif_new(&netif_ppp_config);
+    assert(s_netif);
+#if CONFIG_NET_CONNECT_PPP_DEVICE_USB
+    esp_netif_action_start(s_netif, 0, 0, 0);
+    esp_netif_action_connected(s_netif, 0, 0, 0);
+#else // DEVICE is UART
+    s_stop_task = false;
+    if (xTaskCreate(ppp_task, "ppp connect", 4096, NULL, 5, NULL) != pdTRUE) {
```

Comment:
Should the ppp task stack be configurable like for thread connection?

Reply:
Making PPP task parameters configurable per interface is planned.
 However, this requires a broader architectural redesign of interface initialization and task management, which will be addressed in a future release.